pkgs/development/embedded/openocd-rp2040/default.nix

```diff
@@ -0,0 +1,68 @@
+{ lib
+, stdenv
+, fetchgit
+, pkg-config
+, hidapi
+, libftdi1
+, libusb1
+, which
+, libtool
+, autoconf
+, automake
+, texinfo
+, git
+, libgpiod
+}:
+
+stdenv.mkDerivation rec {
+  pname = "openocd-rp2040";
+  version = "0.11.0";
+  src = fetchgit {
+    url = "https://github.com/raspberrypi/openocd";
+    rev = "228ede43db3665e470d2e518730de013a8c74411";
+    sha256 = "sha256-NrN01BdkLlM9nHilpOSarJvlDWFCVMVInLP/l7GoCIc=";
+    leaveDotGit = true;
+  };
+
+  nativeBuildInputs = [ pkg-config ];
+
+  buildInputs = [ hidapi libftdi1 libusb1 which libtool autoconf automake texinfo git ]
+    ++ lib.optional stdenv.isLinux libgpiod;
+
+  configureFlags = [
+    "--enable-picoprobe"
+  ];
+
+  configurePhase = ''
+    ./bootstrap
+    ./configure --prefix=$out --enable-picoprobe
+  '';
+
+  postInstall = lib.optionalString stdenv.isLinux ''
+    mkdir -p "$out/etc/udev/rules.d"
+    rules="$out/share/openocd/contrib/60-openocd.rules"
+    if [ ! -f "$rules" ]; then
+        echo "$rules is missing, must update the Nix file."
+        exit 1
+    fi
```

Comment:
that check seems useless for running the package since there should always be rules files as long as no one updates the package. Maybe add a note in the long description instead telling users to use `services.udev.packages` for this package instead?

Reply:
I'm doing this to match https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/embedded/openocd/default.nix

They have these lines of code and they also make sense for the rp2040 fork, so I would be inclined to do it the same way as they do. In general I would like to copy their code as much as possible, since the openocd derivation is better maintained than this openocd-rp2040 one.

Reply:
fair enough